cypress/support/step_definitions/timetable.js

```diff
@@ -1,10 +1,57 @@
-import { And, Then } from 'cypress-cucumber-preprocessor/steps';
+import { And, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import { activity, appointment } from '../../mockApis/data';
+
+import { format, addDays } from 'date-fns';
 
 Then('I am displayed todays timetable', () => {
   cy.get('.todays-events > .govuk-body').contains(/.+/);
   cy.log('Timetable data has been passed through');
 });
 
-And('click the {string}', timetableLink => {
-  cy.get('a').contains(timetableLink).first().click();
+const daysFromNow = n => format(addDays(new Date(), n), 'yyyy-MM-dd');
+
+const tableToObject = table => {
+  const [keys, ...rows] = table.rawTable;
+  return rows.map(values =>
+    keys.reduce((o, k, i) => {
```

Comment:
I think this may have been partially copied from elsewhere. 

I would have tended to use `acc, k ,i`, with `acc` being the accumulator.

I wouldn't usually do single letter variables unless its very abstract / mathematical